### PR TITLE
fix emoji detection issue

### LINKED ISSUE #108: fix isOnlyEmoji so that numbers are not considered emoji
the `isOnlyEmoji` function currently has a bug. it detects numbers (e.g. `123`) as an emoji. this leads to a bug:

- #99

a test case has been added:

https://github.com/creatorsgarten/kaogeek-discord-bot/blob/2c204aa04871b0a3e527d781e76e1f8119e9b0e0/test/spec/utils/isOnlyEmoji.spec.ts#L24-L26

please fix the function so that this test passes

### PR DESCRIPTION
**Pull request by 01110111-wave**
# Description
fix #108 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshot
image

# Checklist:

- [x] I have run `pnpm format` and my code don't have any linting issues
  
  
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/utils/isOnlyEmoji.ts [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
 export default (msg: string): boolean => {
   const emoji = msg.match(emojiRegex)
+  //the issue of #99(https://github.com/creatorsgarten/kaogeek-discord-bot/pull/99) only happend with only number message,
+  //so detect number in message and return false if there is.
+  if (emoji !== null) {
+    const unicoded = emoji.map((emo) => {
+      return emo.codePointAt(0)?.toString(16)
+    })
+    for (const i of unicoded) {
+      if (i !== undefined && isNumber(i)) {
```
Comment (ExitedState) [inline src/utils/isOnlyEmoji.ts:13]: from this https://github.com/kaogeek/kaogeek-discord-bot/pull/109#discussion_r1202093527 
I am sorry, I missed that case. btw, I think we can add `i.length === 2` to resolve that problem
```suggestion
      if (i !== undefined && i.length === 2 && /^\d+$/.test(i)) {
```
This way will improve performance significantly,
but its fine if you want to use your version. I just suggest it.

File: src/utils/isOnlyEmoji.ts [hunk 2]
```diff
@@ -1,8 +1,35 @@
 const emojiRegex =
-  /(<a?(:\w+:\d+)>|\p{Emoji_Presentation}|\p{Extended_Pictographic}|\p{Emoji_Component})\s*/gu
+  /(<a?(:\w+:\d+)>|\p{Emoji_Presentation}|\p{Extended_Pictographic}|\p{Emoji_Component}|:\w+:\s*)/gu
 
 export default (msg: string): boolean => {
   const emoji = msg.match(emojiRegex)
+  //the issue of #99(https://github.com/creatorsgarten/kaogeek-discord-bot/pull/99) only happend with only number message,
+  //so detect number in message and return false if there is.
```
Comment (ExitedState) [inline src/utils/isOnlyEmoji.ts:7]: Remove unnecessary comments
```suggestion

```

File: src/utils/isOnlyEmoji.ts [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
+    '32',
+    '33',
+    '34',
+    '35',
+    '36',
+    '37',
+    '38',
+    '39',
+  ]
+  return numberUnicode.includes(input)
```
Comment (beam41) [inline src/utils/isOnlyEmoji.ts:34]: I think remove `.toString(16)`
and use
```ts
function isNumber(input: number): boolean {
  return input >= 0x30 && input <= 0x39
```
would works?

Comment (ExitedState) [reply src/utils/isOnlyEmoji.ts:34]: I will be back to these discuss https://github.com/kaogeek/kaogeek-discord-bot/pull/109#discussion_r1200684597

**Review by ExitedState — CHANGES_REQUESTED** [review]

File: src/utils/isOnlyEmoji.ts (hunk 3 above)
Comment (ExitedState) [reply src/utils/isOnlyEmoji.ts]: it is same to these discuss https://github.com/kaogeek/kaogeek-discord-bot/pull/109#discussion_r1201978058 ??

Comment (beam41) [reply src/utils/isOnlyEmoji.ts]: Yeah, I'm just skip `.toString(16)`.
Also comparison with string in js is weird.

**Comment by dtinth** [comment]
- please update the `isOnlyEmoji.spec.ts` to remove the `.todo` so that tests are run

- please test with this emoji to make sure that your new implementation still matches this: 1️⃣

**Review by dtinth — APPROVED** [review]
thank you! all tests are passing now.

although there are rooms for improvement, every reviewed issue is contained within one file, which makes them easy to improve.

gonna merge this in for now and make it better later.

**Pull request merged**